core/vtk/ttkLDistanceMatrix/ttkLDistanceMatrix.cpp

```diff
@@ -0,0 +1,157 @@
+#include <LDistance.h>
+#include <ttkLDistanceMatrix.h>
+
+#include <vtkDataArray.h>
+#include <vtkDoubleArray.h>
+#include <vtkInformationVector.h>
+#include <vtkMultiBlockDataSet.h>
+#include <vtkNew.h>
+#include <vtkObjectFactory.h>
+#include <vtkPointData.h>
+#include <vtkTable.h>
+
+#include <set>
+
+using namespace std;
+using namespace ttk;
+
+vtkStandardNewMacro(ttkLDistanceMatrix);
+
+ttkLDistanceMatrix::ttkLDistanceMatrix() {
+  SetNumberOfInputPorts(1);
+  SetNumberOfOutputPorts(1);
+}
+
+int ttkLDistanceMatrix::FillInputPortInformation(int port,
+                                                 vtkInformation *info) {
+  if(port == 0) {
+    info->Set(vtkDataObject::DATA_TYPE_NAME(), "vtkMultiBlockDataSet");
+    return 1;
+  }
+  return 0;
+}
+
+int ttkLDistanceMatrix::FillOutputPortInformation(int port,
+                                                  vtkInformation *info) {
+  if(port == 0) {
+    info->Set(vtkDataObject::DATA_TYPE_NAME(), "vtkTable");
+    return 1;
+  }
+  return 0;
+}
+
+// to adapt if your wrapper does not inherit from vtkDataSetAlgorithm
+int ttkLDistanceMatrix::RequestData(vtkInformation * /*request*/,
+                                    vtkInformationVector **inputVector,
+                                    vtkInformationVector *outputVector) {
+  Timer tm{};
+
+  const auto blocks = vtkMultiBlockDataSet::GetData(inputVector[0], 0);
+
+  if(blocks == nullptr) {
+    return 0;
+  }
+
+  const size_t nInputs{blocks->GetNumberOfBlocks()};
+
+  // Get input data
+  std::vector<vtkImageData *> inputData(nInputs);
+
+  for(size_t i = 0; i < nInputs; ++i) {
+    inputData[i] = vtkImageData::SafeDownCast(blocks->GetBlock(i));
+  }
+
+  // sanity check: data non null and same data size
+  {
+    std::set<vtkIdType> sizes{};
+    for(size_t i = 0; i < nInputs; ++i) {
+      if(inputData[i] == nullptr) {
+        this->printErr("Some input block is not a vtkImageData");
```

Comment:
are you restricting to vtkImageData?
(this should be generic vtkDataSet)

Reply:
I've updated the code to be more generic. vtkDataSet are now supported here.